src/serac/physics/solid_mechanics.hpp

```diff
@@ -31,39 +31,33 @@ namespace solid_mechanics {
  * systems of linear equations that show up in implicit
  * solid mechanics simulations
  */
-const IterativeSolverOptions default_linear_options = {.rel_tol     = 1.0e-6,
-                                                       .abs_tol     = 1.0e-16,
-                                                       .print_level = 0,
-                                                       .max_iter    = 500,
-                                                       .lin_solver  = LinearSolver::GMRES,
-                                                       .prec        = HypreBoomerAMGPrec{}};
+const LinearSolverOptions default_linear_options = {.linear_solver  = LinearSolver::GMRES,
+                                                    .preconditioner = Preconditioner::HypreAMG,
+                                                    .relative_tol   = 1.0e-6,
+                                                    .absolute_tol   = 1.0e-16,
+                                                    .max_iterations = 500,
+                                                    .print_level    = 0};
 
 /// the default direct solver option for solving the linear stiffness equations
-const DirectSolverOptions direct_linear_options = {.print_level = 0};
+const LinearSolverOptions direct_linear_options = {.linear_solver = LinearSolver::SuperLU, .print_level = 0};
 
 /**
  * @brief default iteration limits, tolerances and verbosity for solving the
  * systems of nonlinear equations that show up in implicit
  * solid mechanics simulations
  */
 const NonlinearSolverOptions default_nonlinear_options = {
-    .rel_tol = 1.0e-4, .abs_tol = 1.0e-8, .max_iter = 10, .print_level = 1};
+    .relative_tol = 1.0e-4, .absolute_tol = 1.0e-8, .max_iterations = 10, .print_level = 1};
 
-/// the default linear and nonlinear solver options for (quasi-)static analyses
-const SolverOptions default_static_options = {default_linear_options, default_nonlinear_options};
-
-/// solver options that use a direct linear solver for (quasi-)static analyses
-const SolverOptions direct_static_options = {direct_linear_options, default_nonlinear_options};
-
-/// the default solver and time integration options for dynamic analyses
-const SolverOptions default_dynamic_options = {
-    default_linear_options, default_nonlinear_options,
-    TimesteppingOptions{TimestepMethod::Newmark, DirichletEnforcementMethod::RateControl}};
+/**
+ * @brief default quasistatic timestepping options for solid mechanics
+ */
```

Comment:
I think I used the /** */ method mostly because my doxygen plug in defaults to that. I can make a pass at changing this, but it might be best to be in a separate PR for clarity. I'll change this one though.